docs/Iniciando com React/5-React Hooks.md

```diff
@@ -0,0 +1,87 @@
+# React Hooks
+
+## O que são Hooks
+Hooks são funções utilizadas para ter acesso recursos de **estado** e **ciclo de vida**, em **componentes funcionais**. Eles não podem ser utilizados em componentes de classe. Os Hooks no React são nomeados com o prefixo 'use'.
+
+### ⚠ Aviso:
+> Não use Hooks dentro de loops, regras condicionais ou funções aninhadas (funções dentro de funções). Em vez disso, sempre use Hooks no nível superior de sua função React. Seguindo essas regras, você **garante que os Hooks serão chamados na mesma ordem** a cada vez que o componente renderizar.
+
+---
+
+## useState
+O *useState* é um dos hooks utilizados para criar estado dentro de um componente em React. O estado criado pelo *useState* nada mais é que uma variável que o componente consegue reagir quando seu valor é atualizado.
+
+O *useState* recebe como parâmetro o valor inicial do estado. E retorna uma *array* onde o primeiro valor é o estado atual, e o segundo valor é uma função usada para alterar o valor do estado.
+
+### Exemplo:
+```jsx
+import React, { useState } from 'react';
+
+function Contador () {
+  // Cria estado 'clicks' com valor inicial 0
+  // Um jeito fácil de separar o retorno do useState é fazendo desestruturação de array
+  const [clicks, setClicks] = useState(0);
+  
+  return(
+    <div>
+      <h1>{clicks}</h1>
+      {/* A função 'setClicks' é usada para mudar valor de 'clicks' */}
+      <button onClick={() => setClicks(clicks + 1)}>
+        Clique Aqui!
+      </button>
+    </div>
+  );
+}
+```
+
+---
+
+## useEffect
+O *useEffect* é um Hook utilizado para realizar ações quando alguma de suas dependências muda de valor. Esse Hook recebe respectivamente como parâmetros: uma função de *callback* e uma *array* de dependências.
```

Comment:
Seria **um** array ou **uma** array?

Reply:
@EduSantosBrito não é nenhum dos dois, estrangerismo não tem gênero quando vindo do inglês, é só questão de constume, mas minha defesa de "uma array". 

Array = matriz, uma matriz, é estranho você mudar o gênero da parada se for estrangeirismo para mim.

Reply:
> @EduSantosBrito não é nenhum dos dois, estrangerismo não tem gênero quando vindo do inglês, é só questão de constume, mas minha defesa de "uma array".
> 
> Array = matriz, uma matriz, é estranho você mudar o gênero da parada se for estrangeirismo para mim.

Isso que você falou não tem sentido algum! Não é por que um é sinônimo do outro que deve ser falado no feminino, só por que a outra palavra é feminino. Cada palavra possui seu gênero, e quando a palavra é indefinido, geralmente é utilizado o gênero masculino.

De qualquer forma, cada pessoa costuma usar o gênero que prefere para palavras que vem do inglês. Por exemplo, você prefere falar _uma array_ e eu um prefiro falar _um array_, e ambos estão certos. Mas, se der um search no Google por "um array" x "uma array", verá que o gênero masculino ganha de disparado para esse caso.

Independente do que eu acho melhor ou você acha melhor, o mais correto é tentar não usar nenhum dos dois. Nesse caso, poderia ser: **Esse Hook recebe respectivamente como parâmetros: função de callback e array de dependências.** (_uma_ está só fazendo enfeite nessa frase)

Reply:
@gleidsonh você só repetiu o que eu falei

> cada pessoa costuma usar o gênero que prefere para palavras que vem do inglês
> estrangerismo não tem gênero quando vindo do inglês

Aonde está a discordância?

> e ambos estão certos

Aonde eu discordei disso?

E o paralelo facilita muito para introduzir qualquer pessoa ao outro idioma, é extremamente relevante manter o gênero igual na palavra em português e estrangeirismo, por que o overhead mental é menor, mudar o gênero de palavras não é tão simples quanto parece, existe um custo de comunicação bem grande a ponto de well você estar tendo uma discussão sobre isso.

Sobre ser mais popular, não falei que não é. Nem que é, faço menor ideia qual é mais popular.

> Esse Hook recebe respectivamente como parâmetros: função de callback e array de dependências

Deixou de ser "o" para "esse" todos eles estão em um padrão, para que mudar o padrão? 

E por favor, não use exclamação, é só feio mesmo.

Reply:
A referência do @EduSantosBrito é válida. Na documentação da mozilla, *array* é tratado como um tipo de objeto, logo "o objeto array" -> "o array". Fui mais afundo nisso, e lembrei que *array* também pode ser traduzido para "arranjo" ou "vetor" (https://pt.wikipedia.org/wiki/Arranjo_(computa%C3%A7%C3%A3o)).

Estava convencido do argumento do @EduardoRFS, mas pensei melhor com a argumentação do @gleidsonh, e acho que faz mais sentido utilizar o linguajar comum, do que traduzir a palavra.

Reply:
Concordo em usar o mais popular tb